src/Joblify.Core.Data/UnitOfWork/UnitOfWork.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Joblify.Core.Data.Context;
+using Joblify.Core.Data.Models;
+using Joblify.Core.Data.Repository;
+using Joblify.Core.Data.Repository.Interfaces;
+using Microsoft.EntityFrameworkCore;
+
+namespace Joblify.Core.Data.UnitOfWork
+{
+    public class UnitOfWork : IUnitOfWork
+    {
+        private readonly JoblifyDbContext _dbContext;
+        public IRepository<User> UserRepository => new GenericRepository<User>(_dbContext);
+        public IRepository<File> FileRepository => new GenericRepository<File>(_dbContext);
+        public IRepository<Role> RoleRepository => new GenericRepository<Role>(_dbContext);
+        public IRepository<ExternalProvider> ExternalProviderRepository => new GenericRepository<ExternalProvider>(_dbContext);
```

Comment:
Do we want to create new instance for literally every access? 
Consider something like:
```suggestion
 public IRepository<User> UserRepository => _userRepository ?? (_userRepository = new GenericRepository<User>(_context));
```
where _userRepository would be explicitly typeof GenericRepository